src/providers/hana/qgshanautils.cpp

```diff
@@ -0,0 +1,505 @@
+    if ( uri.hasParam( QStringLiteral( "sslCryptoProvider" ) ) )
+      addItem( "sslCryptoProvider", uri.param( QStringLiteral( "sslCryptoProvider" ) ) );
+    if ( uri.hasParam( QStringLiteral( "sslValidateCertificate" ) ) )
+      addItem( "sslValidateCertificate", uri.param( QStringLiteral( "sslValidateCertificate" ) ) );
+    if ( uri.hasParam( QStringLiteral( "sslHostNameInCertificate" ) ) )
+      addItem( "sslHostNameInCertificate", uri.param( QStringLiteral( "sslHostNameInCertificate" ) ) );
+    if ( uri.hasParam( QStringLiteral( "sslKeyStore" ) ) )
+      addItem( "sslKeyStore", uri.param( QStringLiteral( "sslKeyStore" ) ) );
+    if ( uri.hasParam( QStringLiteral( "sslTrustStore" ) ) )
+      addItem( "sslTrustStore", uri.param( QStringLiteral( "sslTrustStore" ) ) );
+  }
+
+  return connectionItems.join( QStringLiteral( " " ) );
+}
+
+QString QgsHanaUtils::quotedIdentifier( const QString &str )
+{
+  QString result = str;
+  result.replace( '"', QStringLiteral( "\"\"" ) );
+  return result.prepend( '\"' ).append( '\"' );
+}
+
+QString QgsHanaUtils::quotedString( const QString &str )
+{
+  QString result = str;
+  result.replace( '\'', QStringLiteral( "''" ) );
+  return result.prepend( '\'' ).append( '\'' );
+}
+
+QString QgsHanaUtils::quotedValue( const QVariant &value )
+{
+  if ( value.isNull() )
+    return QStringLiteral( "NULL" );
+
+  switch ( value.type() )
+  {
+    case QVariant::Int:
+    case QVariant::LongLong:
+    case QVariant::Double:
+      return value.toString();
+    case QVariant::Bool:
+      return value.toBool() ? QStringLiteral( "TRUE" ) : QStringLiteral( "FALSE" );
+    case QVariant::String:
+    default:
+      return quotedString( value.toString() );
+  }
+}
+
+QString QgsHanaUtils::toConstant( const QVariant &value, QVariant::Type type )
+{
+  if ( value.isNull() )
+    return QStringLiteral( "NULL" );
+
+  switch ( type )
+  {
+    case QVariant::Bool:
+      return value.toBool() ? QStringLiteral( "TRUE" ) : QStringLiteral( "FALSE" );
+    case QVariant::Int:
+    case QVariant::UInt:
+    case QVariant::LongLong:
+    case QVariant::ULongLong:
+    case QVariant::Double:
+      return value.toString();
+    case QVariant::Char:
+    case QVariant::String:
+      return QgsHanaUtils::quotedString( value.toString() );
+    case QVariant::Date:
+      return QStringLiteral( "date'%1'" ).arg( value.toDate().toString( "yyyy-MM-dd" ) );
+    case QVariant::DateTime:
+      return QStringLiteral( "timestamp'%1'" ).arg( value.toDateTime().toString( "yyyy-MM-dd hh:mm:ss.zzz" ) );
+    case QVariant::Time:
+      return QStringLiteral( "time'%1'" ).arg( value.toTime().toString( "hh:mm:ss.zzz" ) );
+    case QVariant::ByteArray:
+      return QStringLiteral( "x'%1'" ).arg( QString( value.toByteArray().toHex() ) );
+    default:
+      return value.toString();
+  }
+}
+
+QString QgsHanaUtils::toString( QgsUnitTypes::DistanceUnit unit )
+{
+  switch ( unit )
+  {
+    case QgsUnitTypes::DistanceMeters:
```

Comment:
this looks like a duplicate of QgsUnitTypes code

Reply:
The strings returned here are keys to entries in the database's `ST_UNITS_OF_MEASURE` view. The strings returned by `QgsUnitTypes::encodeUnit` are different, e.g. `meters` instead of `meter` or `km` instead of `kilometer`.

Reply:
Fair enough! Could you add a comment with this explanation?

Reply:
Done in d101482